sources/imdsclient/src/lib.rs

```diff
@@ -400,19 +442,29 @@ mod error {
     #[derive(Debug, Snafu)]
     #[snafu(visibility = "pub(super)")]
 
+    // snafu doesn't yet support the lifetimes used by std::sync::PoisonError
     pub enum Error {
         #[snafu(display("Response '{}' from '{}': {}", get_status_code(source), uri, source))]
         BadResponse { uri: String, source: reqwest::Error },
 
+        #[snafu(display("Fetched IMDSv2 session token"))]
+        EmptyToken,
+
         #[snafu(display("IMDS fetch failed after {} attempts", attempt))]
         FailedFetchIMDS { attempt: u8 },
 
-        #[snafu(display("Failed to fetch IMDSv2 session token after {} attempts", attempt))]
-        FailedFetchToken { attempt: u8 },
+        #[snafu(display("Failed to fetch IMDSv2 session token"))]
+        FailedFetchToken,
+
+        #[snafu(display("Failed to read token within ImdsClient"))]
```

Comment:
Nit: Since these are "internal" implementation detail errors, let's make the message a bit clearer.  Something like "Failed to read token from RwLock..."? (I'm not sold on that wording in particular)